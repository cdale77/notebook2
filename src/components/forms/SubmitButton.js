import React from "react";
import PropTypes from "prop-types";

const SubmitButton = ({ value, onSubmit, errors }) => {
  const errorsPresent = errors => {
    return Object.keys(errors).length > 0 ? true : false;
  };

  return (
    <div className="form-group">
      <input
        className="button"
        name="submit"
        id="submit"
        type="submit"
        value={value || "Submit"}
        onClick={onSubmit}
        disabled={errorsPresent(errors)}
      />
    </div>
  );
};

SubmitButton.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  value: PropTypes.string,
  errors: PropTypes.object
};

SubmitButton.defaultProps = {
  errors: {}
};
export default SubmitButton;
