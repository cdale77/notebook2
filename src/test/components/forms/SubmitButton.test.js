import React from "react";
import Renderer from "react-test-renderer";
import TestUtils from "react-dom/test-utils";
import TestWrapper from "../../helpers/TestWrapper";
import SubmitButton from "../../../components/forms/SubmitButton";

describe("SubmitButton", () => {
  describe("rendering", () => {
    it("should render a button", () => {
      const component = Renderer.create(
        <SubmitButton value="Login" onSubmit={jest.fn()} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should have a default value", () => {
      const component = Renderer.create(<SubmitButton onSubmit={jest.fn()} />);

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should be disabled if errors are present", () => {
      const errors = {
        firstName: "required."
      };
      const component = Renderer.create(
        <SubmitButton onSubmit={jest.fn()} errors={errors} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should be enabled if errors are not present", () => {
      const errors = {};
      const component = Renderer.create(
        <SubmitButton onSubmit={jest.fn()} errors={errors} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("submitting", () => {
    it("should fire the submit callback when clicked", () => {
      const callBack = {
        onSubmit: () => {}
      };

      const spy = jest.spyOn(callBack, "onSubmit");
      const component = TestUtils.renderIntoDocument(
        <TestWrapper>
          <SubmitButton onSubmit={callBack.onSubmit} />
        </TestWrapper>
      );
      const button = TestUtils.findRenderedDOMComponentWithClass(
        component,
        "button"
      );

      TestUtils.Simulate.click(button);

      expect(spy).toHaveBeenCalled();
      expect(callBack.onSubmit.mock.calls.length).toEqual(1);
    });
  });
});
