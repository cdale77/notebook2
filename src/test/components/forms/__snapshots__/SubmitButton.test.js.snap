// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton rendering should be disabled if errors are present 1`] = `
<div
  className="form-group"
>
  <input
    className="button"
    disabled={true}
    id="submit"
    name="submit"
    onClick={[Function]}
    type="submit"
    value="Submit"
  />
</div>
`;

exports[`SubmitButton rendering should be enabled if errors are not present 1`] = `
<div
  className="form-group"
>
  <input
    className="button"
    disabled={false}
    id="submit"
    name="submit"
    onClick={[Function]}
    type="submit"
    value="Submit"
  />
</div>
`;

exports[`SubmitButton rendering should have a default value 1`] = `
<div
  className="form-group"
>
  <input
    className="button"
    disabled={false}
    id="submit"
    name="submit"
    onClick={[Function]}
    type="submit"
    value="Submit"
  />
</div>
`;

exports[`SubmitButton rendering should render a button 1`] = `
<div
  className="form-group"
>
  <input
    className="button"
    disabled={false}
    id="submit"
    name="submit"
    onClick={[Function]}
    type="submit"
    value="Submit"
  />
</div>
`;
